src/core/backbone_router/local.cpp

```diff
@@ -298,6 +347,48 @@ void Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
     }
 }
 
+void Local::ApplyMeshLocalPrefix(void)
+{
+    VerifyOrExit(IsEnabled(), OT_NOOP);
+
+    Get<ThreadNetif>().UnsubscribeMulticast(mAllNetworkBackboneRouters);
+    mAllNetworkBackboneRouters.GetAddress().SetMulticastNetworkPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix());
+    Get<ThreadNetif>().SubscribeMulticast(mAllNetworkBackboneRouters);
+
+    if (IsPrimary())
+    {
+        Get<ThreadNetif>().RemoveUnicastAddress(mBackboneRouterPrimaryAloc);
+        mBackboneRouterPrimaryAloc.GetAddress().SetPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix());
+        Get<ThreadNetif>().AddUnicastAddress(mBackboneRouterPrimaryAloc);
+    }
+
+exit:
+    return;
+}
+
+void Local::UpdateAllDomainBackboneRouters(Leader::DomainPrefixState aState)
+{
+    if (!IsEnabled())
+    {
+        Get<ThreadNetif>().UnsubscribeMulticast(mAllDomainBackboneRouters);
+        ExitNow();
+    }
+
+    if (aState == Leader::kDomainPrefixRemoved || aState == Leader::kDomainPrefixRefreshed)
+    {
+        Get<ThreadNetif>().UnsubscribeMulticast(mAllDomainBackboneRouters);
+    }
+
+    if (aState == Leader::kDomainPrefixAdded || aState == Leader::kDomainPrefixRefreshed)
+    {
+        mAllDomainBackboneRouters.GetAddress().SetMulticastNetworkPrefix(*Get<Leader>().GetDomainPrefix());
```

Comment:
If `mAllDomainBackboneRouters` is only set up with correct prefix when BBR is enabled, should the method `const Ip6::Address &GetAllDomainBackboneRoutersAddress(void) const` return `OT_ERROR_INVALID_STATE` when BBR is not enabled?

Reply:
we have already had the same discussion in #4795 
https://github.com/openthread/openthread/pull/4795#discussion_r407312005